lib/pdk/module/update.rb

```diff
@@ -81,27 +86,27 @@ def describe_ref_to_s(describe_ref)
       end
 
       def new_template_version
-        PDK::Util.default_template_ref
+        metadata_template_uri.git_ref
```

Comment:
This (and the corresponding CLI) needs to honour `--template-ref` otherwise if the module is generated against a static ref like a tag or a commit, running `pdk update` will be updating the module to the same ref it is already at.